.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+      - develop
+      - feat/dev-245-build-binaries
+    tags:
+      - "*"
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - 3.5
+          - 3.6
+          - 3.7
+    env:
+      GDC_CLIENT_ZIP: "gdc-client.zip"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Run Tests
+      if: matrix.os != 'windows-latest'
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+        tox -e py
+    - name: Package
+      run: |
+        cd bin
+        pip install virtualenv
+        . ./package
+        echo ::set-env name=GDC_CLIENT_ZIP::$GDC_CLIENT_ZIP
+      shell: bash
+    - uses: actions/upload-artifact@v2
+      with:
+        name: ${{ env.GDC_CLIENT_ZIP }}-py${{ matrix.python }}.zip
+        path: bin/gdc-client_*.zip
```

Comment:
Does this commit the build artifact into the repository?  If so, that can cause some bloat in our repo.  Git is not a good place to store binary data.  Are there actions that would allow us to upload the artifacts some where else?  Or even to a different repo?

Each build is ~45 MB in total.   We don't do that many dtt releases.  So, if we committed these for each release, we'd slowly grow the repos size.  16 releases over five years.  That's ~150 MB per year.  Over time, it'll add up though.

But if we committed artifacts for every merge to `develop`, then we'll grow much faster.  395 commits over 5 years.  That 3.5 GB per year.

Reply:
The artifacts are not committed to the repo, they are uploaded to a public repository for public repos on github, See https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts for more details and expiry

Reply:
We could do long term storage for artifacts on Nexus. I thought we might originally build on Jenkins and store on Nexus prior to this PR. I'm not entirely sure how that process would look like though.